src/ape/managers/project/types.py

```diff
@@ -194,8 +194,15 @@ def create_manifest(
                 name=self.name,
                 version=self.version,
             )
+            del manifest.contract_types
             # Cache the updated manifest so `self.cached_manifest` reads it next time
             self.manifest_cachefile.write_text(manifest.json())
+            manifest.contract_types = contract_types
+            if manifest.contract_types:
+                for name, contract_type in manifest.contract_types.items():
+                    (self.project_manager.local_project._cache_folder / f"{name}.json").write_text(
```

Comment:
What happens if the file already exists here?

Reply:
now only newly. compiled files would be written

Reply:
Updates the cached copy, which is what we want (driving updates based on compilation changes)